### PR TITLE
fix phpstan level 2 warnings

### PR DESCRIPTION
**Pull request by clxmstaab**
- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [x] Changelog updated?

      Please, update CHANGELOG.md by running next command:
      $ php bin/changelog

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -2,7 +2,7 @@ includes:
     - test/phpstan-baseline.neon
 
 parameters:
-    level: 2
+    level: 3
```
Comment (clxmstaab) [inline phpstan.neon:5]: fixed level 2 .. now targeting level 3 for the next PR

File: src/Component/PharUpdate/Version/Builder.php [hunk 2]
```diff
@@ -14,18 +14,14 @@ class Builder extends Version
 {
     /**
      * Removes the build metadata identifiers.
-     *
-     * @return Builder The Version builder.
      */
     public function clearBuild()
     {
-        return $this->build = array();
```
Comment (clxmstaab) [inline src/Component/PharUpdate/Version/Builder.php:22]: in this case I wasn't sure what todo.

the documented return type did not match the implemenation.
not sure the return type (and the return) is required at all.

need advice.

File: test/phpstan-baseline.neon [hunk 3]
```diff
@@ -11,17 +11,12 @@ parameters:
...
 
 		-
-			message: "#^Method Deployer\\\\Component\\\\PharUpdate\\\\Version\\\\Builder\\:\\:clearPreRelease\\(\\) should return Deployer\\\\Component\\\\PharUpdate\\\\Version\\\\Builder but return statement is missing\\.$#"
+			message: "#^Property Deployer\\\\Component\\\\PharUpdate\\\\Update\\:\\:\\$file \\(string\\) does not accept null\\.$#"
 			count: 1
-			path: ../src/Component/PharUpdate/Version/Builder.php
+			path: ../src/Component/PharUpdate/Update.php
 
 		-
 			message: "#^Default value of the parameter \\#2 \\$code \\(string\\) of method Deployer\\\\Support\\\\Changelog\\\\ParseException\\:\\:__construct\\(\\) is incompatible with type int\\.$#"
```
Comment (clxmstaab) [inline test/phpstan-baseline.neon:19]: I guess this one is a phpstan bug.

reported upstream: https://github.com/phpstan/phpstan/issues/4011

File: src/Component/PharUpdate/Version/Builder.php (hunk 2 above)
Comment (staabm) [reply src/Component/PharUpdate/Version/Builder.php:22]: Is this code called anywhere?

Comment (antonmedv) [reply src/Component/PharUpdate/Version/Builder.php:22]: ¯\_(ツ)_/¯

We don't have `dep self-update` tests. So, who knows?

**Pull request merged**